Src/VimApp/VimAppHost.cs

```diff
@@ -69,6 +72,25 @@ public override void VimCreated(IVim vim)
             _vim.VimData.CurrentDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
         }
 
+        public override void Close(ITextView textView)
+        {
+            foreach (var vimWindow in _vimWindowManager.VimWindowList)
+            {
+                foreach (var viewInfo in vimWindow.VimViewInfoList.ToList())
+                {
+                    if (viewInfo.TextView == textView)
+                    {
+                        viewInfo.VimBuffer.Close();
+                        viewInfo.TextView.Close();
+                        _viewMap.Remove(viewInfo.TextView);
+                        break;
+                    }
+                }
+                vimWindow.Clear();

```

Comment:
Why do we need to clear the windows here?

Reply:
I was using VimApp to test things like `:e #`, :e file.txt`, `:tabe file.txt`, `:q`,  and especially navigating to marks in a closed buffer. A lot of the `VimHost` was unimplemented and I tried to fix it up so most of those things work like they work in Visual Studio.

I admit I wasn't too worried about getting it just perfect because:

1) It didn't used to work at all
2) VimApp is mostly just used by developers

It is intriguing to think of really investing in VimApp enough to make it competitive with gvim for non-Visual Studio editing!

Reply:
I reverted my vim app changes for the most part, since my implementation was not fully thought out.